### PR TITLE
Docs IPv8 features, Why does IPv8 exist?

### PR DESCRIPTION
**Pull request by synctext**
adds a few new sections to the main index.
Specifically designed to overcome the problem that the docs look empty and never had any love. It is designed to signal that each link click on details will be rewarded with a wealth of welcome information.

### CODE REVIEW

**Comment by qstokkink** [comment]
Completely manual PR check, here are your compilation errors:

```
/home/quinten/Documents/py-ipv8/doc/index.rst:16: ERROR: Title underline too short.

Why does IPv8 exist?
==================
/home/quinten/Documents/py-ipv8/doc/index.rst:16: ERROR: Title underline too short.

Why does IPv8 exist?
==================
```

**Comment by synctext** [comment]
Manually changed 
```
Why does IPv8 exist?
==================
```
into
```
Why does IPv8 exist?
====================
```

File: doc/index.rst [hunk 1]
```diff
@@ -1,8 +1,23 @@
...
+
+Why does IPv8 exist?
+====================
+
+Problems with the very fabric of The Internet, IPv4, are mounting. The approach of IPv6, Mobile IP, and IPSec is hampered by fundamental architectural problems. One solution is moving the intelligence up to a higher layer in the protocol stack and towards the end points. Our end-points are not dependant only any central infrastructure. Our architecture is academically pure and fully decentralised; to the point of obsession. IPv8 is based on the principle of self-governance. Like Bitcoin and Bittorrent our IPv8 overlay is permissionless and requires no maintenance.
```
Comment (devos50) [inline doc/index.rst]: > Our end-points are not dependant only any central infrastructure.

This sentence seems to be grammatically incorrect?

File: doc/index.rst [hunk 2]
```diff
@@ -1,8 +1,23 @@
...
+
+Why does IPv8 exist?
+====================
+
+Problems with the very fabric of The Internet, IPv4, are mounting. The approach of IPv6, Mobile IP, and IPSec is hampered by fundamental architectural problems. One solution is moving the intelligence up to a higher layer in the protocol stack and towards the end points. Our end-points are not dependant only any central infrastructure. Our architecture is academically pure and fully decentralised; to the point of obsession. IPv8 is based on the principle of self-governance. Like Bitcoin and Bittorrent our IPv8 overlay is permissionless and requires no maintenance. 
+
+IPv8 aims to help restore the original Internet; for free; owned by nobody; for everyone. By design IPv8 does not need any committee of oversight, no monthly payment to any essential infrastructure provider and lack any central identity registration service provider.
```
Comment (devos50) [inline doc/index.rst]: > and lack any central identity registration service provider. 

and lack**s**?

File: doc/index.rst [hunk 3]
```diff
@@ -1,8 +1,23 @@
 IPv8 Documentation
 ==================
 
-Welcome to the IPv8 documentation!
-This file will help you navigate the existing documentation.
+Welcome to the IPv8 documentation! This file will help you navigate the existing documentation.
+The documentation will explain the concepts behind IPv8, the architecture, and guide you to making your first overlay.
+
+IPv8 Features
```
Comment (qstokkink) [inline doc/index.rst]: I spotted some partial truths and some incorrect claims regarding the IPv8 functionality. Some of the claims are only true for Tribler and others are only true for the TunnelCommunity:

- `encrypted network connection`  〰️ 
- `to a known person` 〰️ 
- `robust against various network problems and security issues` 〰️
- `is a networking layer` ✔️ 
- `which offers identities` ✔️ 
- `exposes reputation functions` ❌ 
- `storage of reputation data` ❌ 
- `calculates reputation levels`  ❌ 
- `IPv8 is a mechanism designed to build trust`  ❌
- `offers public-key based identities` ✔️ 
- `offers protection against eavesdropping`   〰️
- `man-in-the-middle attacks` 〰️
- `peer failure` ❌
- `network failure` 〰️
- `packet loss` ❌
- `change of IP numbers, network mobility` ✔️ 
- `blocking by NAT/Firewalls` ✔️

File: doc/index.rst (hunk 1 above)
Comment (synctext) [reply doc/index.rst]: fixed in new commit

File: doc/index.rst (hunk 2 above)
Comment (synctext) [reply doc/index.rst]: good catch

File: doc/index.rst (hunk 3 above)
Comment (synctext) [reply doc/index.rst]: Very handy ToDo list, thnx! 
Addressed both all errors and also all half truths.

File: doc/index.rst [hunk 4]
```diff
@@ -1,8 +1,31 @@
...
+
+IPv8 aims to help restore the original Internet; for free; owned by nobody; for everyone. By design IPv8 does not need any committee of oversight, no monthly payment to any essential infrastructure provider and lacks any central identity registration service provider. 
+
+IPv8 Example
+============
+
+IPv8 is a tool to build interesting applications. Several IPv8 overlays can be used to offer various services. The particular application below uses seven IPv8 overlays for server-less discovery, DHT-based lookup, Tor-inspired privacy, decentralised marketplace, etc. For each overlay it shows statistics such as active neighbors and used network traffic.
```
Comment (qstokkink) [inline doc/index.rst]:
```suggestion
IPv8 is a tool to build interesting distributed applications. IPv8 overlays can be used to offer various services for your application. Our flagship application Tribler, for example, uses seven IPv8 overlays for serverless discovery. Tribler's services range from DHT-based lookup, Tor-inspired privacy to a completely decentralised marketplace for bandwidth. If we take a look at Tribler's debug screen, it shows IPv8 in action: you can observe statistics such as active neighbors and utilized network traffic to make sure you have made a healthy overlay.
```

File: doc/index.rst [hunk 5]
```diff
@@ -1,8 +1,31 @@
...
+
+ .. image:: ./further-reading/resources/healthy_IPv8_overlay_collection.png
+   :target: ./further-reading/resources/healthy_IPv8_overlay_collection.png
+   :alt: Alt Text
```
Comment (qstokkink) [inline doc/index.rst]:
```suggestion
   :alt: A screenshot of Tribler's IPv8 statistics
```

File: doc/index.rst [hunk 6]
```diff
@@ -1,8 +1,31 @@
 IPv8 Documentation
 ==================
 
-Welcome to the IPv8 documentation!
-This file will help you navigate the existing documentation.
+Welcome to the IPv8 documentation! This file will help you navigate the existing documentation.
+The documentation will explain the concepts behind IPv8, the architecture, and guide you to making your first overlay.
```
Comment (qstokkink) [inline doc/index.rst]:
```suggestion
The documentation will explain the concepts behind IPv8, the architecture, and guide you through making your first overlay.
```

File: doc/index.rst [hunk 7]
```diff
@@ -1,8 +1,31 @@
...
+
+IPv8 Features
+==================
```
Comment (qstokkink) [inline doc/index.rst]:
```suggestion
=============
```

File: doc/index.rst [hunk 8]
```diff
@@ -1,8 +1,31 @@
...
+
+It offers public-key based identities and offers protection against man-in-the-middle attacks, some network failures, change of IP numbers, network mobility, and blocking by NAT/Firewalls. IPv8 offers global connectivity through integrated UDP NAT puncturing, announcement of your identity claim and a web-of-trust. IPv8 has an integrated attestation service. You can use IPv8 for official verification that something is true or authentic, according to a trustworthy attestor. By using zero-knowledge proofs we attempt to minimize privacy leakage.
+
+Why does IPv8 exist?
```
Comment (qstokkink) [inline doc/index.rst]: It might be best to start with this and then delve into the features we offer.

File: doc/index.rst [hunk 9]
```diff
@@ -1,8 +1,31 @@
...
 
-Welcome to the IPv8 documentation!
-This file will help you navigate the existing documentation.
+Welcome to the IPv8 documentation! This file will help you navigate the existing documentation.
+The documentation will explain the concepts behind IPv8, the architecture, and guide you to making your first overlay.
+
+IPv8 Features
+==================
+
+The IPv8 overlay offers network connection to a known digital identity which is robust against several network problems and security issues. When your external IPv4 network address changes, IPv8 keeps the network connectivity going. IPv8 offers communication based on public keys, it uses a custom NAT-traversing DHT to find their current IPv4 network address. The overlay keeps track of a number of neighbors and occasionally checks if they are still responsive. IPv8 is a networking layer which offers identities, communication with some robustness, and provides hooks for higher layers. For instance, our Internet-deployed reputation functions and ledger-based storage of reputation data. IPv8 in combination with our Trustchain is designed as a mechanism to build trust.
```
Comment (qstokkink) [inline doc/index.rst]:
```suggestion
IPv8 is a networking layer which offers identities, communication with some robustness, and provides hooks for higher layers. For instance, our Internet-deployed reputation functions and ledger-based storage of reputation data. In combination with our Trustchain, IPv8 is designed as a mechanism to build trust.  Each network overlay offers network connections to known digital identities through public keys. Overlays are robust against several network problems and security issues. Using a custom NAT-traversing DHT to find the current IPv4 network address, IPv8 keeps the network connectivity going even as the IPv4 addresses change.  Each network overlay keeps track of a number of neighbors and occasionally checks if they are still responsive.
```

File: doc/index.rst [hunk 10]
```diff
@@ -1,8 +1,31 @@
...
+
+It offers public-key based identities and offers protection against man-in-the-middle attacks, some network failures, change of IP numbers, network mobility, and blocking by NAT/Firewalls. IPv8 offers global connectivity through integrated UDP NAT puncturing, announcement of your identity claim and a web-of-trust. IPv8 has an integrated attestation service. You can use IPv8 for official verification that something is true or authentic, according to a trustworthy attestor. By using zero-knowledge proofs we attempt to minimize privacy leakage.
```
Comment (qstokkink) [inline doc/index.rst]: This seems mostly like a repeat of the last paragraph. Only the last 3 sentences offer something new: strong identity.

File: doc/index.rst [hunk 11]
```diff
@@ -1,8 +1,31 @@
...
+
+Why does IPv8 exist?
+====================
+
+Problems with the very fabric of The Internet, IPv4, are mounting. The approach of IPv6, Mobile IP, and IPSec is hampered by fundamental architectural problems. One solution is moving the intelligence up to a higher layer in the protocol stack and towards the end points. Our end-points are not dependant on any central infrastructure. Our architecture is academically pure and fully decentralised; to the point of obsession. IPv8 is based on the principle of self-governance. Like Bitcoin and Bittorrent our IPv8 overlay is permissionless and requires no maintenance.
```
Comment (qstokkink) [inline doc/index.rst]:
```suggestion
Problems with the very fabric of The Internet, IPv4, are mounting. The approach of IPv6, Mobile IP, and IPSec is hampered by fundamental architectural problems. One solution is moving the intelligence up to a higher layer in the protocol stack and towards the end points. Our endpoints are not dependent on any central infrastructure. Our architecture is academically pure and fully decentralized; to the point of obsession. IPv8 is based on the principle of self-governance. Like Bitcoin and Bittorrent our IPv8 overlays are permissionless and requires no server upkeep costs. 
```

File: doc/index.rst [hunk 12]
```diff
@@ -1,8 +1,31 @@
...
+
+Why does IPv8 exist?
+====================
+
+Problems with the very fabric of The Internet, IPv4, are mounting. The approach of IPv6, Mobile IP, and IPSec is hampered by fundamental architectural problems. One solution is moving the intelligence up to a higher layer in the protocol stack and towards the end points. Our end-points are not dependant on any central infrastructure. Our architecture is academically pure and fully decentralised; to the point of obsession. IPv8 is based on the principle of self-governance. Like Bitcoin and Bittorrent our IPv8 overlay is permissionless and requires no maintenance. 
+
+IPv8 aims to help restore the original Internet; for free; owned by nobody; for everyone. By design IPv8 does not need any committee of oversight, no monthly payment to any essential infrastructure provider and lacks any central identity registration service provider.
```
Comment (qstokkink) [inline doc/index.rst]: Maybe drop the second sentence. Just ending with this single line, sends a very strong message:

> IPv8 aims to help restore the original Internet; for free; owned by nobody; for everyone.

**Comment by qstokkink** [comment]
The science part can pretty much be a copy-paste of the work mentioned in https://github.com/Tribler/tribler/wiki#nat-traversal-80-success-rate and https://github.com/Tribler/tribler/wiki#overlay-protocol-for-synchronization .

**Comment by qstokkink** [comment]
@synctext mind if I hijack this PR and finish it?

**Comment by qstokkink** [comment]
retest this please

**Comment by qstokkink** [comment]
retest this please

**Comment by qstokkink** [comment]
retest this please

**Comment by qstokkink** [comment]
retest this please

**Comment by qstokkink** [comment]
retest this please

**Comment by qstokkink** [comment]
retest this please

**Comment by qstokkink** [comment]
retest this please

**Comment by qstokkink** [comment]
retest this please

**Comment by qstokkink** [comment]
retest this please

**Comment by qstokkink** [comment]
retest this please

**Comment by qstokkink** [comment]
retest this please

**Comment by qstokkink** [comment]
For future reference: you can also select "add to batch" and then commit all of the suggested changes as one big change.

**Comment by synctext** [comment]
```Allow edits from maintainers``` is very much on :-)
Having trouble going through the code review process, learned an important lesson!

File: doc/index.rst [hunk 13]
```diff
@@ -1,9 +1,49 @@
...
+
+IPv8 aims to help restore the original Internet; for free; owned by nobody; for everyone.
+
+
+IPv8 Features
+=============
+
+IPv8 is a networking layer which offers identities, communication with some robustness, and provides hooks for higher layers. For instance, our Internet-deployed reputation functions and ledger-based storage of reputation data. In combination with our Trustchain, IPv8 is designed as a mechanism to build trust.  Each network overlay offers network connections to known digital identities through public keys. Overlays are robust against several network problems and security issues. Using a custom NAT-traversing DHT to find the current IPv4 network address, IPv8 keeps the network connectivity going even as the IPv4 addresses change.  Each network overlay keeps track of a number of neighbors and occasionally checks if they are still responsive.
```
Comment (devos50) [inline doc/index.rst]: going, even as ...

(add the comma)

File: doc/index.rst [hunk 14]
```diff
@@ -1,9 +1,49 @@
...
+
+IPv8 offers global connectivity through integrated UDP NAT puncturing, announcement of your identity claim and a web-of-trust. IPv8 has an integrated attestation service. You can use IPv8 for official verification that something is true or authentic, according to a trustworthy attestor. By using zero-knowledge proofs we attempt to minimize privacy leakage.
```
Comment (devos50) [inline doc/index.rst]: Suggestion: we talk a lot about components here. We could add links to the relevant directories implementing this component?

Comment (qstokkink) [reply doc/index.rst]: I don't have readthedocs documenting the actual source code, so I can't really add simple links to the appropriate classes.

**Pull request merged**